src/components/field.jsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import withApp from './with-app';
+import withModel from './with-model';
+import withLayout from './with-layout';
+
+import './field.scss';
+
+function fieldCounts(dimInfo, field, onlyBar) {
+  let str = '';
+  if (onlyBar) {
+    str = `${dimInfo.qStateCounts.qSelected
+      + dimInfo.qStateCounts.qOption} of ${dimInfo.qCardinal}`;
+  } else if (field.qnPresentDistinctValues !== field.qnTotalDistinctValues) {
+    str = `${dimInfo.qStateCounts.qSelected
+      + dimInfo.qStateCounts.qOption} of ${field.qnTotalDistinctValues}(${
+      field.qnPresentDistinctValues
+    })`;
+  } else {
+    str = `${dimInfo.qStateCounts.qSelected
+      + dimInfo.qStateCounts.qOption} of ${field.qnPresentDistinctValues}`;
+  }
+
+  return str;
+}
+
+export class Field extends React.Component {
+  render() {
+    const {
+      field, fieldData, onlyBar, layout,
+    } = this.props;
+    if (!layout) {
+      return null;
+    }
+
+    const total = layout.qListObject.qDimensionInfo.qCardinal;
+    const states = layout.qListObject.qDimensionInfo.qStateCounts;
+    const green = { width: `${Math.ceil((states.qSelected / total) * 100)}%` };
+    const grey = { width: `${Math.ceil((states.qExcluded / total) * 100)}%` };
+    let classes = `field ${fieldData.qKeyType}`;
+
+    let descriptions = '';
+    if (states.qSelected) {
+      classes += ' filtered';
+    }
+
+    if (fieldData.qHasDuplicates) {
+      classes += ' has-duplicates';
+      descriptions += '. Duplicate values';
+    } else {
+      descriptions += '. Unique values';
+    }
+
+    if (fieldData.qHasNull) {
+      classes += ' has-null';
+      descriptions += ', has nulls';
+    } else {
+      descriptions += ', no nulls.';
+    }
+
+    const name = (
+      <div className="name">
+        <div>{field}</div>
+      </div>
+    );
+
+    const bar = (
+      <div
+        className="gwg"
```

Comment:
Took me a while to figure out what gwg meant. 'state' might be a better class name?

Reply:
It's still a prototype and we are likely going to change most of the css anyway so keep it for now.